src/components/MazeMap/MazeMap.tsx

```diff
@@ -131,17 +133,74 @@ const MazeMap = (props: MazeMapProps) => {
   const prepareMap = () => {
     if (window.Mazemap) {
       const map = new window.Mazemap.Map(mapOptions);
-      map.on('load', () => {
+      map.on("load", () => {
         if (props.controls) {
           map.addControl(new window.Mazemap.mapboxgl.NavigationControl());
         }
+        if (props.hideIcons && (!props.hideAll)) {
```

Comment:
please move this behaviour into another function like the other prop-related stuff in here